js/DrupalImageEditor.es6.js

```diff
@@ -1,179 +1,148 @@
+/*globals Drupal:false */
 /*globals JSONEditor:false */
+/*globals jQuery:false */
 /**
  * @file DrupalImageEditor class.
  *
  * @external Drupal
  *
  * @external JSONEditor
+ *
+ * @external jQuery
  */
 
-export var DrupalImageEditor = JSONEditor.AbstractEditor.extend({
-  getNumColumns: function () {
+export let DrupalImageEditor = JSONEditor.AbstractEditor.extend({
+  getNumColumns: function() {
     return 4;
   },
-
-  build: function () {
+  build: function() {
     this.title = this.header = this.label = this.theme.getFormInputLabel(this.getTitle(), this.isRequired());
+
+    // Editor options
+    this.options = jQuery.extend({}, {
+      'title': 'Browse',
+      'icon': '',
+      'image_url': '/'
+    }, JSONEditor.defaults.options.drupal_image || {}, this.options.drupal_image || {});
+
     // Don't show uploader if this is readonly
-    if (!this.schema.readOnly && !this.schema.readonly) {
-      this.urlfield = this.theme.getFormInputField('text');
-      let media_library_opener_parameters = {
-        field_widget_id: this.urlfield.id
-      };
-      let opener_encoded = encodeURIComponent(JSON.stringify(media_library_opener_parameters));
-      this.urlfield.addEventListener('change', function (e) {
+    if(!this.schema.readOnly && !this.schema.readonly) {
+      this.input = this.theme.getFormInputField('text');
+      this.button = this.getButton(this.path + '-media', 'upload', Drupal.t('Select/Upload Media'));
+      // @todo: Add support for multiple file/image URL editors.
+      let media_library_settings = 'media_library_opener_id=patternkit.opener.jsonlibrary' +
```

Comment:
This is never re-assigned, should be 'const' rather than 'let'.

Reply:
oops.exe